brian2/devices/cpp_standalone/device.py

```diff
@@ -1400,20 +1401,32 @@ def network_run(self, net, duration, report=None, report_period=10*second,
                 for codeobj in obj._code_objects:
                     code_objects.append((obj.clock, codeobj))
 
+        # #Code for time format function
+        # format_func_code = '''
+        #  '''
+
         # Code for a progress reporting function
         standard_code = '''
         void report_progress(const double elapsed, const double completed, const double start, const double duration)
         {
+            std::string start_format = format_time((int)start);
```

Comment:
The start time of the simulation is the "biological time", and should therefore not be formatted in the same way as the elapsed/estimated "wall time". It will often be in seconds or even milliseconds.

Reply:
Noted.